tests/test_datasets/test_hierarchical_dataset.py

```diff
@@ -136,7 +149,150 @@ def product_level_df():
         {
             "timestamp": ["2000-01-01", "2000-01-02"] * 4,
             "segment": ["a"] * 2 + ["b"] * 2 + ["c"] * 2 + ["d"] * 2,
-            "target": [1, 2] + [10, 20] + [100, 200] + [1000, 2000],
+            "target": [1, 1] + [0, 1] + [3, 18] + [7, 2],
+        }
+    )
+    df = TSDataset.to_dataset(df)
+    return df
+
+
+@pytest.fixture
+def product_level_df_w_nans():
```

Comment:
Actually it is better to change the entire fixtures(don't add w_nans prefix) as it is a general case. NaNs in the begging occur almost always. Sorry for suggesting you creating the separate)